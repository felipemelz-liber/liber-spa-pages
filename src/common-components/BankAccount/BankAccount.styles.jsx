import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  flex-direction: row;
`;

export const ImagePlaceHolder = styled.div`
  width: 120px;
  height: 120px;
  border-radius: 4px;
  border: solid 1px #cdd7df;
  padding: 1px;
`;

export const ImageLogo = styled.img`
  width: 120px;
  height: 120px;
  border-radius: 4px;
  border: solid 1px #cdd7df;
  padding: 1px;
`;

export const InfoContainer = styled.div`
  display: flex;
  flex-direction: column;
  padding: 16px;

  font-family: Roboto;
  font-size: 16px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.38;
  letter-spacing: 0.06px;
  text-align: left;
  color: #6287a7;
`;

export const Title = styled.div`
  font-family: Roboto;
  font-size: 16px;
  font-weight: bold;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.38;
  letter-spacing: 0.06px;
  text-align: left;
  color: #6287a7;
  margin-bottom: 8px;
  max-width: 408px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
`;
