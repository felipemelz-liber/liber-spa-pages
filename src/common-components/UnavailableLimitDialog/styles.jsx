import styled from 'styled-components';
import { FlatButton } from 'liber-components';

const TEXT_COLOR = '#6287a7';

export const Container = styled.div`
  padding: 24px 24px 12px 24px;
  width: 480px;

  * {
    font-family: Roboto;
  }
`;

export const Header = styled.div`
  font-size: 21px;
  font-weight: bold;
  line-height: 1.4;
  letter-spacing: 0.08px;
  color: ${TEXT_COLOR};
  margin-bottom: 24px;
`;

export const Content = styled.div`
  font-size: 16px;
  line-height: 1.38;
  letter-spacing: 0.01px;
  margin-bottom: 24px;
  color: ${TEXT_COLOR};
  margin-bottom: 24px;
`;

export const ActionsRow = styled.div`
  display: flex;
  justify-content: flex-end;
`;

export const Button = styled(FlatButton).attrs(() => ({ size: 'large' }))``;
