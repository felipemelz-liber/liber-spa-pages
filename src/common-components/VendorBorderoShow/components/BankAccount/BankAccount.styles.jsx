import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  padding: 2px 0px;
`;

export const ImageContainer = styled.div`
  width: 120px;
  height: 120px;
  object-fit: contain;
  border-radius: 4px;
  border: solid 1px #cdd7df;
  background-color: #f7f7f7;
  margin-right: 16px;
`;

export const Content = styled.div`
  display: flex;
  flex-direction: column;
`;

export const Title = styled.div`
  font-family: Roboto;
  font-size: 16px;
  font-weight: bold;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.38;
  letter-spacing: 0.06px;
  text-align: left;
  color: #6287a7;
  margin-bottom: 8px;
  margin-top: 16px;
`;

export const Description = styled.div`
  font-family: Roboto;
  font-size: 16px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.38;
  letter-spacing: 0.06px;
  text-align: left;
  color: #6287a7;
`;
