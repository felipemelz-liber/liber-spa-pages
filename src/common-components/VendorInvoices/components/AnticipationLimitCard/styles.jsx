import styled from 'styled-components';

export const Card = styled.div`
  font-family: Roboto;
  color: #6287a7;
  width: 360px;
  box-shadow: 0px 3px 6px #00000029;
  border: 1px solid #cdd7df;
  border-radius: 8px;
  padding: 16px 24px;
  margin-bottom: 56px;

  * {
    font-family: Roboto;
  }
`;

export const CardTitle = styled.div`
  font-size: 16px;
  line-height: 22px;
  letter-spacing: 0.06px;
  font-weight: 500;
  color: #4b6f85;
  margin-bottom: 4px;
`;
