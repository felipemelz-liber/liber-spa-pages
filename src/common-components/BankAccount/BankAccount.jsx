import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { Container, ImagePlaceHolder, ImageLogo, InfoContainer, Title } from './BankAccount.styles';
import defaultImage from '../../../../assets/images/bank-logos/default.png';

const formatDigit = digit => (digit && digit !== '' ? `-${digit}` : '');

export const BankAccount = ({ bankCode, bankName, branch, branchCd, account, accountCd }) => {
  const [image, setImage] = useState(null);

  useEffect(() => {
    import(`../../../../assets/images/bank-logos/banco_0${bankCode}.png`)
      .then(res => {
        setImage(res.default);
      })
      .catch(() => {
        setImage(defaultImage);
      });
  }, []);

  return (
    <Container>
      {image ? <ImageLogo src={image} alt={`Logo do ${bankName}`} /> : <ImagePlaceHolder />}
      <InfoContainer>
        <Title>{`${bankCode} - ${bankName}`}</Title>
        <div>{`AgÃªncia ${branch}${formatDigit(branchCd)}`}</div>
        <div>{`Conta ${account}${formatDigit(accountCd)}`}</div>
      </InfoContainer>
    </Container>
  );
};

BankAccount.propTypes = {
  branch: PropTypes.string,
  branchCd: PropTypes.string,
  account: PropTypes.string,
  accountCd: PropTypes.string,
  bankCode: PropTypes.string,
  bankName: PropTypes.string,
};

BankAccount.defaultProps = {
  branch: '',
  branchCd: '',
  account: '',
  accountCd: '',
  bankCode: '',
  bankName: '',
};

export default BankAccount;
