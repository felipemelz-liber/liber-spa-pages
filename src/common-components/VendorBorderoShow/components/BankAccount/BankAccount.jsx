import React from 'react';
import PropTypes from 'prop-types';
import { Container } from './BankAccount.styles';
import { BankAccount as BankAccountComponent } from '../../../BankAccount/BankAccount';

const BankAccount = props => {
  return (
    <Container>
      <BankAccountComponent {...props} />
    </Container>
  );
};

BankAccount.propTypes = {
  branch: PropTypes.string,
  branchCd: PropTypes.string,
  account: PropTypes.string,
  accountCd: PropTypes.string,
  bankName: PropTypes.string,
  bankCode: PropTypes.string,
};

BankAccount.defaultProps = {
  branch: '',
  branchCd: '',
  account: '',
  accountCd: '',
  bankName: '',
  bankCode: '',
};

export default BankAccount;
