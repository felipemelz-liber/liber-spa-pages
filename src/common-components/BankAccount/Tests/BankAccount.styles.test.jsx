import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import * as styles from '../BankAccount.styles';

describe('BankAccount styled components tests', () => {
  configure({ adapter: new Adapter() });

  Object.keys(styles).forEach(style => {
    const Component = styles[style];
    it(`should render ${style} correctly`, () => {
      const wrapper = shallow(<Component />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
