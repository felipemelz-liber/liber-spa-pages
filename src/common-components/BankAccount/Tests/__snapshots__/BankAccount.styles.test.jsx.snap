// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BankAccount styled components tests should render Container correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  display: flex;
  flex-direction: row;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`BankAccount styled components tests should render ImageLogo correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-htpNat",
        "isStatic": false,
        "rules": Array [
          "
  width: 120px;
  height: 120px;
  border-radius: 4px;
  border: solid 1px #cdd7df;
  padding: 1px;
",
        ],
      },
      "displayName": "styled.img",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-htpNat",
      "target": "img",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`BankAccount styled components tests should render ImagePlaceHolder correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": false,
        "rules": Array [
          "
  width: 120px;
  height: 120px;
  border-radius: 4px;
  border: solid 1px #cdd7df;
  padding: 1px;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`BankAccount styled components tests should render InfoContainer correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bxivhb",
        "isStatic": false,
        "rules": Array [
          "
  display: flex;
  flex-direction: column;
  padding: 16px;

  font-family: Roboto;
  font-size: 16px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.38;
  letter-spacing: 0.06px;
  text-align: left;
  color: #6287a7;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bxivhb",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`BankAccount styled components tests should render Title correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-ifAKCX",
        "isStatic": false,
        "rules": Array [
          "
  font-family: Roboto;
  font-size: 16px;
  font-weight: bold;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.38;
  letter-spacing: 0.06px;
  text-align: left;
  color: #6287a7;
  margin-bottom: 8px;
  max-width: 408px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-ifAKCX",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
